### PR TITLE
[MLIR][DISC] Add LhloFusionInliner Pass

### PR DESCRIPTION
**Pull request by linearhit**
This pass inlines the body contents of lmhlo.fusion_op. 
Typically it should work after the body has been lowered into loops. For disc, it works after LhloLegalizeRootsToParallelLoops &  InputInlineFusion in #50102 .

### CODE REVIEW

**Review by joker-eph — COMMENTED** [review]
I think you'll need a CMake change as well

**Comment by linearhit** [comment]
> I think you'll need a CMake change as well

Done. Thanks, i missed it.

**Review by joker-eph — COMMENTED** [review]
LG otherwise

File: tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_fusion_inliner.cc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#define GEN_PASS_CLASSES
+#include "mlir-hlo/Dialect/mhlo/transforms/lmhlo_passes.h.inc"
+
+struct LhloFusionInlinerPass
+    : public LhloFusionInlinerPassBase<LhloFusionInlinerPass> {
+ public:
+  void runOnFunction() override {
+    auto func = getFunction();
+    SmallVector<FusionOp, 4> worklist;
```
[the remark this answers is not in the document]
Comment (linearhit) [reply tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_fusion_inliner.cc:33]: done.

File: tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_fusion_inliner.cc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    func.walk([&](FusionOp fusion) { worklist.push_back(fusion); });
+    for (FusionOp fusion : worklist) {
+      InlineFusion(fusion);
+      fusion.erase();
+    }
+  }
+
+ private:
+  void InlineFusion(FusionOp fusion) {
+    fusion.dump();
```
[the remark this answers is not in the document]
Comment (linearhit) [reply tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_fusion_inliner.cc:43]: done

**Review by joker-eph — APPROVED** [review]

**Pull request merged**